### PR TITLE
Resource service sambo

### PR DESCRIPTION
**Pull request by SamHerrera88**
please revise.

### CODE REVIEW

**Comment by SamHerrera88** [comment]
resolved the problem

File: src/app/game/buildings/buildings.service.ts [hunk 1]
```diff
@@ -91,6 +91,12 @@ export class BuildingsService {
     localStorage.setItem('buildings', JSON.stringify(buildings));
   }
 
+  loadFinishedBuildingsFromLS() {
```
Comment (balazsbarni) [inline src/app/game/buildings/buildings.service.ts]: theese buildings are not finished, so their name should be buildings, loadbuildingsFromLS...

File: src/app/game/resources/resources.component.ts [hunk 2]
```diff
@@ -1,27 +1,73 @@
...
-      });
+    this.resourceService.getResources().subscribe(response => {
+      this.food = response[0].amount;
+      this.money = response[1].amount;
+      this.foodgeneration = response[0].generation;
+      this.moneygeneration = response[1].generation;
+    });
+  }
+
+  generationGenerator() {
```
Comment (balazsbarni) [inline src/app/game/resources/resources.component.ts]: instead maybe updateResources?

File: src/app/game/resources/resources.component.ts [hunk 3]
```diff
@@ -1,27 +1,73 @@
...
+    let generateFood = this.getFoodGeneration();
+    let generatedGold = this.getGoldGeneration();
+    let troopSustenance = this.getSustenaceOfTroops();
+    this.food += generateFood - troopSustenance;
+    this.money += generatedGold;
+  }
+
+  getGoldGeneration() {
+    const completedMines = this.buildingService.loadFinishedBuildingsFromLS();
+    let filteringMinesFromBuildings = completedMines.filter(
```
Comment (balazsbarni) [inline src/app/game/resources/resources.component.ts]: completedMines maybe

File: src/app/game/resources/resources.component.ts [hunk 4]
```diff
@@ -1,27 +1,73 @@
...
+      building => building.finishedAt <= Date.now() && building.type === "mine"
+    );
+    let numberofCompletedMines = filteringMinesFromBuildings.length;
+    return numberofCompletedMines;
+  }
+
+  getSustenaceOfTroops() {
+    const fetchedTroops = this.troopsService.loadTroopsFromLS();
```
Comment (balazsbarni) [inline src/app/game/resources/resources.component.ts]: just troops

File: src/app/game/resources/resources.component.ts [hunk 5]
```diff
@@ -1,27 +1,73 @@
...
+  generationGenerator() {
+    let generateFood = this.getFoodGeneration();
+    let generatedGold = this.getGoldGeneration();
+    let troopSustenance = this.getSustenaceOfTroops();
+    this.food += generateFood - troopSustenance;
+    this.money += generatedGold;
+  }
+
+  getGoldGeneration() {
+    const completedMines = this.buildingService.loadFinishedBuildingsFromLS();
```
Comment (balazsbarni) [inline src/app/game/resources/resources.component.ts]: should use consistently gold or money

File: .vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (balazsbarni) [inline .vscode/launch.json]: whats that?

**Review by balazsbarni — CHANGES_REQUESTED** [review]

**Review by balazsbarni — APPROVED** [review]

**Pull request merged**